ekncontent/ekncontent/eknc-domain.c

```diff
@@ -820,6 +908,16 @@ on_xapian_query_response (GObject *source,
       return;
     }
 
+  state->models = g_ptr_array_new_full (state->total_models, maybe_unref_object);
+  g_ptr_array_set_size (state->models, state->total_models);
```

Comment:
Not quite - https://developer.gnome.org/glib/stable/glib-Pointer-Arrays.html#g-ptr-array-new-full does not actually set the *size* of the model, all it does is preallocate memory. The size will still be zero.